vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -247,18 +237,29 @@ def _get_server_instance(self, instance):
                                    tags=service_check_tags, message=err_msg)
                 raise Exception(err_msg)
 
-            self.server_instances[i_key] = server_instance
+            return server_instance
+
+    def _get_server_instance(self, instance):
+        i_key = self._instance_key(instance)
+        tags = instance.get('tags', [])
+
+        service_check_tags = [
+            'vcenter_server:{0}'.format(instance.get('name')),
+            'vcenter_host:{0}'.format(instance.get('host')),
+        ] + tags
+        service_check_tags = list(set(service_check_tags))
+
+        if i_key not in self.server_instances:
+            self.server_instances[i_key] = self._smart_connect(instance, service_check_tags)
 
         # Test if the connection is working
         try:
-            self.server_instances[i_key].RetrieveContent()
+            self.server_instances[i_key].CurrentTime()
```

Comment:
Could we have a for loop with a number of attempts instead and maybe an exponential decreasing delay between each attempt? 

We could even make this logic a `with` statement I guess. What do you think?

Reply:
I don't think we need more than one retry here. It is just in case the connection happens to expire, so that the cached server instance can be refreshed. If it is a real failure, we can wait for the next check run.

Reply:
ok. I hope you are right.